# HG changeset patch
# User David Bateman <dbateman@free.fr>
# Date 1206022658 -3600
#      Thu Mar 20 15:17:38 2008 +0100
# Node ID 25eacc0c27060e67c737c0314885d2f7d81b6116
# Parent  52f90c7adac6c745071bd8627160de15bc8af087
Ensure k is an integer scalar in rotdim

diff --git a/scripts/ChangeLog b/scripts/ChangeLog
--- a/scripts/ChangeLog
+++ b/scripts/ChangeLog
@@ -1,10 +1,12 @@
 2008-03-20  David Bateman  <dbateman@free.fr>
 
+	* general/rotdim.m: Ensure k is an integer scale.
+
 	* general/circshift.m: If matrix is empty fast return.
 
 2008-03-20  Jaroslav Hajek  <highegg@localhost.localdomain>
 
 	* linear-algebra/subspace.m: Check number of arguments and number
 	of dimensions of arguments.
 
 2008-03-19  Jaroslav Hajek  <highegg@gmail.com>
diff --git a/scripts/general/rotdim.m b/scripts/general/rotdim.m
--- a/scripts/general/rotdim.m
+++ b/scripts/general/rotdim.m
@@ -53,18 +53,18 @@
 
 function y = rotdim (x, k, plane)
   
   if (nargin < 1 || nargin > 3)
     print_usage ();
   endif
 
   if (nargin > 1 && ! isempty(k))
-    if (imag (k) != 0 || fix (k) != k)
-      error ("rotdim: k must be an integer");
+    if (!isscalar (k) || imag (k) != 0 || fix (k) != k)
+      error ("rotdim: k must be an scalar integer");
     endif
   else
     k = 1;
   endif
 
   nd = ndims (x);
   sz = size (x);
   if (nargin < 3)
